modules/apps/29-fee/ibc_module.go

```diff
@@ -197,6 +220,15 @@ func (im IBCModule) OnTimeoutPacket(
 	packet channeltypes.Packet,
 	relayer sdk.AccAddress,
 ) error {
-	// TODO: Implement fee specific logic if fee is enabled for the given channel
+	packetId := channeltypes.NewPacketId(packet.SourceChannel, packet.SourcePort, packet.Sequence)
+	identifiedPacketFee, exists := im.keeper.GetFeeInEscrow(ctx, packetId)
+	if !exists {
+		return sdkerrors.Wrapf(types.ErrFeeNotFound, "fee not found for packet id %s", packetId)
+	}
```

Comment:
first check must be `IsFeeEnabled`, we don't want non fee enabled channels erroring on logic that doesn't apply to it